scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -628,14 +638,14 @@ object Execution {
       Trampoline(cache.getOrElseInsert(conf, this, {
         cache.writer match {
           case ar: AsyncFlowDefRunner =>
-            val (fut, cancelHandler) = ar.validateAndRun(conf)(result(_, mode))
-            (fut.map { m => ((), Map(m)) }, cancelHandler)
+            val CFuture(fut, cancelHandler) = ar.validateAndRun(conf)(result(_, mode))
+            CFuture(fut.map { m => ((), Map(m)) }, cancelHandler)
           case other =>
             val fut = Future.failed(
               new IllegalArgumentException(
                 s"requires cascading Mode producing AsyncFlowDefRunner, found mode: $mode and writer ${other.getClass}: $other"))
 
-            (fut, CancellationHandler.empty)
+            CFuture(fut, CancellationHandler.empty)
```

Comment:
maybe `CFuture.failed(ex: Exception): CFuture[Nothing]` here? and move this empty logic there?

Reply:
👍